File: array-partition-i/array-partition-i.cpp
class Solution {
public:
    int arrayPairSum(vector<int>& nums) {
        
        if(nums.empty()) return 0;
        sort(nums.begin(),nums.end()); //sorting the array because to find the minimum pair and then add we need to have the difference the least
        int result=0;
        
        for(int i=0;i<nums.size();i=i+2){
            result+=nums[i];
        }
        return result;
        
    }
};

File: array-partition-i/array-partition-i.py
class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        
        if(len(nums) == 0):
            return 0
        nums.sort()
        sum=0
        b=len(nums)
        
        for a in range(0,b,2):
            sum+=nums[a]
        
        return sum
        
        
        